### PR TITLE
Add Vite

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/889

Summary of the changes made:
1. Craco configuration removed.
2. Vite installed and necessary configurations added.
3. Env variables added in the env fie.
4. ASP NET Core Web api and React app detached.
5. CORS policy added for the BE.
6. Views removed.
7. Redirects and View responses removed.
8. New Docker image file added for the React app , old one refactored to build and run only BE.
9. The React app will be served using nginx and the required configuration for the nginx was done.
10. Necessary github workflows added for CI of the FE . Old ones refactored to match new requirements.
11. Ports adjusted. FE is on 5002 and the BE is on 5010.

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.production [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_UI_SERVER_URL = 'http://217.174.159.212:5010'
+VITE_ADMINISTRATION_URL = 'http://217.174.159.212:5001'
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/.env.production:2]: Shouldn't this be https://alpha.judge.softuni.org and admin.alpha...

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.staging [hunk 2]
```diff
@@ -0,0 +1,2 @@
+VITE_UI_SERVER_URL = 'http://217.174.159.217:5010'
+VITE_ADMINISTRATION_URL = 'http://217.174.159.217:5001'
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/.env.staging:2]: Same here for dev

File: Servers/UI/OJS.Servers.Ui/appsettings.json [hunk 3]
```diff
@@ -9,10 +9,11 @@
   },
   "ApplicationUrls": {
     "AdministrationUrl": "http://localhost:5001",
-    "UiUrl": "http://localhost:5002",
+    "UiUrl": "http://localhost:5010",
     "SulsPlatformBaseUrl": "https://localhost:44304",
     "SulsPlatformApiKey": "1234"
   },
+  "FRONTEND_URL": "http://localhost:5002",
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/appsettings.json:16]: Add this into "ApplicationUrls" section

File: docker-compose.yml [hunk 4]
```diff
@@ -22,6 +22,28 @@ services:
...
       - mq
       - redis
+  fe:
+    build:
+      context: .
+      dockerfile: ./Docker/Dockerfile.fe
+      args:
+        - BUILD_COMMAND=build:local
+    image: judge_web_fe
+    container_name: web_ui_fe_instance
```
Comment (gogo4ds) [inline docker-compose.yml:32]: web_fe_instance

File: Servers/UI/OJS.Servers.Ui/OJS.Servers.Ui.csproj [hunk 5]
```diff
@@ -44,6 +44,8 @@
       <_ContentIncludedByDefault Remove="Views\Shared\_Layout.cshtml" />
       <_ContentIncludedByDefault Remove="Views\Shared\_ValidationScriptsPartial.cshtml" />
       <_ContentIncludedByDefault Remove="ClientApp\dist\asset-manifest.json" />
+      <_ContentIncludedByDefault Remove="Views\Home\Index.cshtml" />
+      <_ContentIncludedByDefault Remove="Views\_ViewImports.cshtml" />
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/OJS.Servers.Ui.csproj:48]: Do you need these here?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.production (hunk 1 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/.env.production:2]: I think this is the Ip of the https://alpha.judge.softuni.org/

image

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.staging (hunk 2 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/.env.staging:2]: Same like the ip's for prod

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.production (hunk 1 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/.env.production]: Done

File: Servers/UI/OJS.Servers.Ui/appsettings.json (hunk 3 above)
Comment (MTsatsarov) [reply Servers/UI/OJS.Servers.Ui/appsettings.json]: Done

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/.env.production [hunk 6]
```diff
@@ -0,0 +1,2 @@
+VITE_UI_SERVER_URL = 'https://217.174.159.212:5010'
+VITE_ADMINISTRATION_URL = 'https://admin.alpha.judge.softuni.org'
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/.env.production:2]: Why one is IP and the other is not?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

**Review by gogo4ds — APPROVED** [review]

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-dev-all.yml [hunk 7]
```diff
@@ -8,6 +8,9 @@ jobs:
     secrets: inherit
   build-and-deploy-ui:
     uses: ./.github/workflows/ui-dev-deploy.yml
+    secrets: inherit
+      build-and-deploy-ui:
```
[the remark this answers is not in the document]
Comment (MTsatsarov) [reply .github/workflows/deploy-dev-all.yml]: Fixed.

**Review by gogo4ds — APPROVED** [review]

**Review by gogo4ds — DISMISSED** [review]

**Review by gogo4ds — DISMISSED** [review]

**Review by gogo4ds — DISMISSED** [review]

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**